### PR TITLE
DATAES-317 - Introduce query logging in ElasticsearchTemplate.

### PR DESCRIPTION
**Pull request by xhaggi**
The java client of elasticsearch does not provide an easy way to log the serialized query (JSON) by setting the log level of some logger to DEBUG. Therefore this PR adds logging capability to the ElasticsearchTemplate because it enclose the ES `SearchRequestBuilder` where you need to call `toString()` to get the real JSON query.

The logging is done by a separate logger `org.springframework.data.elasticsearch.core.QUERY` which needs to be set to log level DEBUG to output the serialized query.

- [x] You have read the [Spring Data contribution guidelines](https://github.com/spring-projects/spring-data-build/blob/master/CONTRIBUTING.adoc).
- [x] There is a ticket in the bug tracker for the project in our [JIRA](https://jira.spring.io/browse/DATAES).
- [x] You use the code formatters provided [here](https://github.com/spring-projects/spring-data-build/tree/master/etc/ide) and have them applied to your changes. Don’t submit any formatting related changes.
- [ ] You submit test cases (unit or integration tests) that back your changes.
- [X] You added yourself as author in the headers of the classes you touched. Amend the date range in the Apache license header if needed. For new types, add the license header (copy from another file and set the current year only).

### CODE REVIEW

File: src/main/java/org/springframework/data/elasticsearch/core/ElasticsearchTemplate.java [hunk 1]
```diff
@@ -931,7 +930,13 @@ private SearchResponse doSearch(SearchRequestBuilder searchRequest, SearchQuery
 				searchRequest.addAggregation(aggregatedFacet.getFacet());
 			}
 		}
-		return getSearchResponse(searchRequest.setQuery(searchQuery.getQuery()).execute());
+		return getSearchResponse(searchRequest.setQuery(searchQuery.getQuery()));
+	}
+
+	private SearchResponse getSearchResponse(SearchRequestBuilder requestBuilder) {
+		if (QUERY_LOGGER.isDebugEnabled())
+			QUERY_LOGGER.debug(requestBuilder.toString());
```
Comment (shawnjohnson) [inline src/main/java/org/springframework/data/elasticsearch/core/ElasticsearchTemplate.java]: I'm not sure if Spring has any guidelines for this, but I found this old document recommending to include curly-braces around if statements - http://www.oracle.com/technetwork/java/javase/documentation/codeconventions-142311.html

File: src/main/java/org/springframework/data/elasticsearch/core/ElasticsearchTemplate.java [hunk 2]
```diff
@@ -136,7 +137,8 @@
  */
 public class ElasticsearchTemplate implements ElasticsearchOperations, ApplicationContextAware {
 
-	private static final Logger logger = LoggerFactory.getLogger(ElasticsearchTemplate.class);
+	private static final Logger QUERY_LOGGER = LoggerFactory.getLogger("org.springframework.data.elasticsearch.core.QUERY");
```
Comment (shawnjohnson) [inline src/main/java/org/springframework/data/elasticsearch/core/ElasticsearchTemplate.java]: I like this idea, though I don't know if there are any conventions within Spring / Spring Data to look towards following for this type of logging?  For example in the JPA project they have the following options for logging queries - http://www.baeldung.com/sql-logging-spring-boot

**Comment by olivergierke** [comment]
I've polished and merged that for Lovelace and Kay.

**Comment by xhaggi** [comment]
@olivergierke thanks 👍

**Pull request closed without merging**